src/Sdk/WebApi/WebApi/BrokerHttpClient.cs

```diff
@@ -103,24 +103,31 @@ public async Task<TaskAgentMessage> GetRunnerMessageAsync(
                 new HttpMethod("GET"),
                 requestUri: requestUri,
                 queryParameters: queryParams,
+                readErrorBody: true,
                 cancellationToken: cancellationToken);
 
             if (result.IsSuccess)
             {
                 return result.Value;
             }
 
-            // the only time we throw a `Forbidden` exception from Listener /messages is when the runner is
-            // disable_update and is too old to poll
-            if (result.StatusCode == HttpStatusCode.Forbidden)
+            if (TryParseErrorBody(result.ErrorBody, out BrokerError brokerError))
             {
-                throw new AccessDeniedException($"{result.Error} Runner version v{runnerVersion} is deprecated and cannot receive messages.")
+                if (result.StatusCode == HttpStatusCode.Forbidden && brokerError.Code == 1)
```

Comment:
In RunServiceError, the `Code` is the HTTP status code. It helps us distinguish the HTTP status code returned by Run Service any intermediate layer.

Here is looks like `Code` means something else. Any reason not to follow the same pattern across our all of services?

Reply:
Along that same line, wondering what `1` indicates. If we need another error code in addition to the HTTP status code, wondering whether a friendly code (i.e. string) would be better.

Reply:
Yeah that's a fair question. The `AccessDeniedException` has a Code=1, which I was going to match. But I guess we're not tied to that.